internal/kafka/internal/handlers/cluster.go

```diff
@@ -108,6 +126,15 @@ func (h clusterHandler) RegisterEnterpriseCluster(w http.ResponseWriter, r *http
 	handlers.Handle(w, r, cfg, http.StatusOK)
 }
 
+func (h clusterHandler) validateOCMProviderAvailable(provider clusters.Provider, err error) handlers.Validate {
+	return func() *errors.ServiceError {
+		if err != nil || shared.IsNil(provider) {
```

Comment:
I'm curious about what shared.IsNil provides over `== nil`?

Reply:
@ziccardi might know more haha

Reply:
The function was born mainly to be used by the array's predicate (IsNilPredicate).
There it was mandatory because we didn't know if the generic `T` would have been a pointer or not.

Then we decided it was useful as a shared public function.

The main advantage over `== nil` should be that `IsNil` does not suffer the `nil is not nil` issue.